lib/vagrant-service-manager/plugin.rb

```diff
@@ -8,6 +11,15 @@ class Plugin < Vagrant.plugin('2')
         require_relative 'command'
         Command
       end
+
+      config 'servicemanager' do
+        require_relative 'config'
+        Config
+      end
+
+      action_hook(:servicemanager, :machine_action_up) do |hook|
```

Comment:
This should only run when a config.service-manager line is in the Vagrantfile, otherwise it runs on every box I start whether it should or not.  The only other way is to detect ADB and CDK which we can't do afaik right now.